baybe/parameters/fidelity.py

```diff
@@ -0,0 +1,269 @@
+        converter=lambda x: cattrs.structure(x, tuple[float, ...]),
+        # FIXME[typing]: https://github.com/python-attrs/attrs/issues/1197
+        validator=[
+            min_len(2),
+            validate_is_finite,
+            deep_iterable(ge(0.0)),
+        ],
+    )
+    """The costs associated with querying the parameter at each value."""
+
+    _target_fidelity: str = field(
+        alias="target_fidelity",
+        converter=str,
+    )
+    """The column index of the target fidelity value."""
+
+    # TODO Jordan MHS: handle hyperparameters from different acqfs.
+    _zeta: tuple[float, ...] | float = field(
+        alias="zeta",
+        # FIXME[typing]: https://github.com/python-attrs/cattrs/issues/111
+        converter=lambda x: (
+            None
+            if x is None
+            else cattrs.structure(x, float)
+            if isinstance(x, (float, int))
+            else cattrs.structure(x, tuple[float, ...])
+        ),
+        default=None,
+    )
+    """The maximum discrepancy from target fidelity at any design choice."""
+
+    @_costs.validator
+    def _validate_cost_length(  # noqa: DOC101, DOC103
+        self, _: Any, value: tuple[float, ...]
+    ) -> None:
+        """Validate that ``len(_costs)`` equals ``len(_values)``.
+
+        Raises:
+            ValueError: If ``len(_costs) != len(_values)``.
+        """
+        if len(value) != len(self._values):
+            raise ValueError(f"Length of 'costs' and 'values' different in {self.name}")
+
+    @_zeta.validator
+    def _validate_zeta(  # noqa: DOC101, DOC103
+        self, _: Any, value: tuple[float, ...] | float
+    ) -> None:
+        """Validate instance attribute ``zeta``.
+
+        Raises:
+            ValueError: If ``zeta`` is ``tuple[float, ...]`` and
+                ``len(zeta) != len(self._values)``.
+            ValueError: If ``zeta`` contains any negative or infinite values.
+        """
+        if value is None:
+            # Jordan MHS TODO: add adaptive_zeta argument
+            # Jordan MHS TODO: handle other kinds of discrepancy param
+            # Jordan MHS TODO: make zeta optional if above params defined
+            raise ValueError(f"Value 'zeta' undefined in {self.name}")
+
+        elif isinstance(value, float):
+            if not np.isfinite(value):
+                raise ValueError(f"Parameter 'zeta' is infinite in {self.name}")
+            elif value < 0:
+                raise ValueError(f"Parameter 'zeta' is negative in {self.name}")
+            else:
+                return
+
+        else:
+            assert isinstance(self._zeta, tuple)
+
+            if len(self._zeta) != len(self._values):
+                raise ValueError(
+                    f"Tuples 'zeta' and 'values' are different lengths in {self.name}"
+                )
+
+            if any(np.isinf(value)):
+                raise ValueError(
+                    f"Tuple 'zeta' contains infinite values in {self.name}"
+                )
```

Comment:
This would all simplify by a lot if we used the conversion step suggested in my other comment. So let's see how we decide (cc: @Scienfitz) and potentially fix this